pkg/advisory/advisory.go

```diff
@@ -168,13 +187,13 @@ func Discover(options DiscoverOptions) error {
 
 			if exists {
 				// TODO: Should we allow for updating existing advisories if previously read vuln
-				// data has been updated?
+				//  data has been updated?
```

Comment:
I think we have to somehow, right? If NVD reclassifies a vuln from LOW->CRITICAL, that needs to be reflected.

Maybe it doesn't show up as `vuln.severity=CRITICAL` (and therefore always has been) but instead as a new advisory entry saying that (and when) the severity changed. Our SLO for fixing the critical vuln starts from when it's marked as critical, not from when it was originally detected as low.

(This doesn't have to happen in this PR, just jotting down ideas about it which we can talk about later)

Reply:
Yeah, this is a great point. 💯  I've made a note of this in a scratchpad doc I'll share with you.

I have a lot of thoughts about how to evolve this "event" data structure to account for situations like this, and I want to get advisory data moved into its own repo, and the structs out of Melange, first.